enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,176 @@
+superseded-by:
+  - ""
+---
+
+# Configuring Webhook Token Authenticators
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+The standard mechanism for external authentication integration is via `tokenreviews.authentication.k8s.io`
+from the kube-apiserver. We will support configuring the kube-apiserver this way and ensure it always
+works by using the same mechanism to drive the standard openshift oauth-server authentication integration.
+
+## Motivation
+
+Using the standard kube-apiserver extension mechanism will allow external integrations like Keycloak or
+any other Kubernetes authentication stack to integrate in a standard way.
+
+`webhookTokenAuthenticators` is a kube-apiserver configuration directive that can be used to
+configure an external authenticator which supports Kubernetes' token reviews (see
+[Webhook Token Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#webhook-token-authentication) for more details).
+By implementing this enhancement, we would significantly simplify any drop-in replacements
+for the integrated oauth-server, which might be helpful in cases where we would like to
+delegate authentication to a different OpenShift control plane, or direct authentication to an
+identity provider capable of handling the Kubernetes token reviews, such as Keycloak.
+
+### Goals
+
+1. Make it possible to configure an endpoint capable of performing kube token reviews
+2. Modify cluster-authentication-operator so that it configures authentication via the OpenShift OAuth
+   server by using the `webhookTokenAuthneticators` field instead of the current patching of the
+   kube-apiserver.
+
+### Non-Goals
+
+1. provide a partial authentication operator to any external authenticator.  Once the standard oauth-server
+   is disabled, authentication configuration is *entirely* the responsibility of the new authentication stack.
+2. allow both the built-in oauth-server and another authentication stack to be enabled at the same time.
+
+## Proposal
+
+### Implementation
+
+#### API
+
+`authentications.config.openshift.io/v1` already contains the `spec.webhookTokenAuthenticators`. It
+expects an array of references to secrets in the `openshift-config` namespace. The field is currently
+unused so it is safe to leverage it for the purposes of configuring webhook token authenticators.
+
+Any secret reference which gets set to this field has to be validated. In 4.4, it will only be
+possible to configure a single element for the `wehbhookTokenAuthenticators` field and the referenced
+kubeconfigs will only be allowed to use the `-data` forms of their fields to configure
```

Comment:
list the exact keys to whitelist.  I'm thinking kubeconfig, ca-bundle.crt, client.key, client.cert

Reply:
I would simply start with the secret being expected to have `kubeconfig` key

Reply:
Or do you mean a specific whitelist of the kubeconfig fields?